### PR TITLE
MULE-11029: Parsers are not validating the presence of a required par…

### PR DESCRIPTION
**Pull request by estebanwasinger**
…ameter

### CODE REVIEW

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java [hunk 1]
```diff
@@ -134,6 +135,12 @@ protected ResolverSet getParametersAsResolverSet(ParameterizedModel model, List<
 
       if (resolver != null) {
         resolverSet.add(parameterName, resolver);
+      } else if (p.isRequired()) {
+        throw new IllegalStateException(format("Parameter '%s' of type %s from the %s '%s' is required but wasn't set",
+                                               parameterName,
+                                               getTypeId(p.getType()).orElse(StringUtils.EMPTY),
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java]: olEse("Unknwown") or something like that

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java [hunk 2]
```diff
@@ -223,6 +230,10 @@ public void resolveParameters(ObjectType objectType, DefaultObjectBuilder builde
 
       if (valueResolver != null) {
         builder.addPropertyResolver(objectField.getName(), valueResolver);
+      } else if (field.isRequired() && !field.getAnnotation(FlattenedTypeAnnotation.class).isPresent()) {
+        throw new IllegalStateException(format("Class %s contains field '%s' of type %s which is required but wasn't set",
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java]: Error message shouldn't contain the terms class or field. From the user's point of view, he's missing a parameter. He doesn't know about the internal structure

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java [hunk 3]
```diff
@@ -223,6 +230,10 @@ public void resolveParameters(ObjectType objectType, DefaultObjectBuilder builde
...
+                                               objectClass.getName(), getLocalPart(field),
+                                               getTypeId(field.getValue()).orElse(StringUtils.EMPTY)));
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java]: unknown

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java (hunk 1 above)
Comment (alegmarra) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java]: use `ExtensionMetadataTypeUtils.getId`, it already handles the optional and a fallback for Java format. Maybe improve it to receive a default instead on it being always `""`

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java [hunk 4]
```diff
@@ -223,6 +230,10 @@ public void resolveParameters(ObjectType objectType, DefaultObjectBuilder builde
 
       if (valueResolver != null) {
         builder.addPropertyResolver(objectField.getName(), valueResolver);
+      } else if (field.isRequired() && !field.getAnnotation(FlattenedTypeAnnotation.class).isPresent()) {
```
Comment (alegmarra) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java]: `ExtensionMetadataTypeUtils#isParameterGroup`

**Pull request merged**